packages/notifier/src/asyncIteratableKit.js

```diff
@@ -0,0 +1,84 @@
+// @ts-check
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
+
+import { HandledPromise, E } from '@agoric/eventual-send';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+import './types';
+
+/**
+ * This makes a pair of an `updater` and an initial `asyncIteratable`. The
+ * `updater` API is the same as the Notifier's `updater`. In both cases,  calls
+ * to the `updater` produce a stream of non-final values terminated with
+ * either a final success value or a final failure reason. The purpose of the
+ * notifier is to be lossy over the non-final values. The purpose of the
+ * asyncIterableKit is to consume such streams losslessly.
+ *
+ * The initial `asyncIteratable` represents the stream starting with the first
+ * update to the `updater`. It makes async iterators each of which advance
+ * independently starting at that starting point. An async iterator has a
+ * `snapshot()` method which will create a new async iterable capturing the
+ * iterator's current position as its starting point.
```

Comment:
I don’t believe this works as advertised as it would imply that the updater would multicast. I would expect to see additional pubsub machinery to fork the stream.

Reply:
I believe we've verbally clarified why it does work. Resolving. But if anything about this is still mysterious, please unresolve.